### ISSUE TITLE
Improve multi-direction scrolling with trackpad

### ISSUE THREAD

**Issue by guidezpl** [flutter/devtools#4159]
Devtools has become a real joy to use, congrats team! :)

I face one regular annoyance however, with scrolling the widget tree (although this applies to anything that could be scrollable in the X and Y axis).

When scrolling diagonally with a trackpad, it feels like handlers are fighting each other to decide whether to scroll vertically or horizontally, leading to a janky experience. There are two possible solutions

1. Only allow scrolling in one direction (vertically), with horizontal scrolling happening automatically
2. (preferred) Allow diagonal scrolling

In this video demonstrating the issue, I'm attempting to scroll sideways and diagonally

https://user-images.githubusercontent.com/6655696/171608203-c60dea5b-c724-44e7-9464-382348d7760f.mp4

**Comment by kenzieschmoll** [comment]
@Piinks @goderbauer on the framework side of things, is there any ongoing work to improve the 2D scroll experience in Flutter that our team should be tracking?

**Comment by Piinks** [comment]
Yes! I have actually been experimenting with a 2D scroll view - that can scroll diagonally. :D
It is likely a long way off from landing yet, being a large new scrolling system, but I am very optimistic so far.
https://github.com/flutter/flutter/issues/87370 is the tracking issue
